src/org/intellij/grammar/generator/ParserGenerator.java

```diff
@@ -780,6 +790,16 @@ else if (javaType != Java.INTERFACE && i == 1) {
     myShortener = shortener;
   }
 
+  @NotNull
+  protected String generatePackageName(String className) {
+    return StringUtil.getPackageName(className);
+  }
+
+  @NotNull
+  protected String adjustName(String importName) {
```

Comment:
`adjustName` must happen in BNF PSI, not in the generator.

As we need to support both "generate-time" and "edit-time" (in-IDE support)